src/wire/ip.rs

```diff
@@ -833,7 +833,10 @@ pub fn pretty_print_ip_payload<T: Into<Repr>>(
                             )?;
                             let valid =
                                 udp_packet.verify_checksum(&repr.src_addr(), &repr.dst_addr());
-                            format_checksum(f, valid)
+                            let partially_valid = udp_packet
+                                .verify_partial_checksum(&repr.src_addr(), &repr.dst_addr());
+
+                            format_checksum(f, valid || partially_valid)
```

Comment:
I'd prefer different formatting for valid vs partially valid checksum, since this is a diagnostic interface.